### ISSUE TITLE
signal received (hangup), next signal will force immediate shutdown

### ISSUE THREAD

**Issue by hyahm** [foxcpp/maddy#203]
# Describe the bug

What do you think is wrong?

# Steps to reproduce

# Log files
```
remote: TLS verify error, trying without authentication {"domain":"163.com","msg_id":"a8cb6e57-5e58f2c6","reason":"x509: certificate is valid for *.163.com, 163.com, not 163mx01.mxmail.netease.com.","remote_server":"163mx01.mxmail.netease.com."}
queue: delivered        {"attempt":0,"msg_id":"a8cb6e57","rcpt":"tlztl1227@163.com"}
submission: authentication failed       {"src_ip":"45.133.99.2:16132","username":"qoubeg@hyahm.com"}
submission: authentication failed       {"src_ip":"78.128.113.66:49802","username":"zrmbxs@hyahm.com"}
submission: authentication failed       {"src_ip":"78.128.113.66:29614","username":"zrmbxs@hyahm.com"}
submission: authentication failed       {"src_ip":"45.133.99.130:20658","username":"zrmbxs@hyahm.com"}
submission: authentication failed       {"src_ip":"45.133.99.2:35872","username":"tqevvw@hyahm.com"}
submission: authentication failed       {"src_ip":"45.133.99.2:9584","username":"cejvs@hyahm.com"}
submission: authentication failed       {"src_ip":"78.128.113.66:5438","username":"zrmbxs@hyahm.com"}
signal received (hangup), next signal will force immediate shutdown.
```
# Configuration file

$(hostname) = mail.caesarvideos.store
$(primary_domain) = mail.caesarvideos.store
$(local_domains) = $(primary_domain)

tls /etc/maddy/certs/$(hostname)/fullchain.pem \
    /etc/maddy/certs/$(hostname)/privkey.pem

sql local_mailboxes local_authdb {
    driver sqlite3
    dsn all.db
}

limits local_limits {
    # Up to 50 msgs/sec across any amount of SMTP connections.
    all rate 50 1s
}

checks local_checks { }

modifiers local_modifiers {
    alias_file /etc/maddy/aliases
}

limits outbound_limits {
    destination rate 20 1s
    destination concurrency 10
}

checks outbound_checks { }

modifiers outbound_modifiers {
    sign_dkim $(primary_domain) default
}

mx_auth outbound_auth {
    dane
    mtasts {
        cache fs
        fs_dir mtasts_cache/
    }
    sts_preload {
        source eff # See https://startls-everywhere.org
        enforce_testing yes
    }
    local_policy {
        min_tls_level encrypted
        min_mx_level none
    }
}

hostname $(hostname)

submission tls://0.0.0.0:465 {
    limits &local_limits
    auth &local_authdb

    source $(local_domains) {
        destination $(local_domains) {
            check &local_checks
            modify &local_modifiers
            deliver_to &local_mailboxes
        }
        default_destination {
            check &outbound_checks
            modify &outbound_modifiers
            deliver_to &remote_queue
        }
    }
    default_source {
        reject 501 5.1.8 "Non-local sender domain"
    }
}

queue remote_queue {
    target remote {
        limits &outbound_limits
        mx_auth &outbound_auth
    }

    autogenerated_msg_domain $(primary_domain)
    bounce {
        destination $(local_domains) {
            check &local_checks
            modify &local_modifiers
            deliver_to &local_mailboxes
        }
        default_destination {
            reject 550 5.0.0 "Refusing to send DSNs to non-local addresses"
        }
    }
}

**Comment by foxcpp** [comment]
Answer the first question in the template. "What do you think is wrong?" I don't see anything strange in the log snippet you provided.

**Comment by foxcpp** [comment]
Feel free to reopen once you add necessary details about your problem.

**Comment by arisudesu** [comment]
Caught this with openmetrics enabled on version 0.6.3. Enabling openmetrics endpoint causes the process to print this line on TERM, but it does not shut down until second TERM. Commenting openmetrics causes clean shutdown on first TERM signal.

I'm sorry for reporting for outdated version, I didn't have time to check/update my configs for latest version yet.

maddy.conf that I use:
```
$(hostname) = mydomain.tld
$(primary_domain) = mydomain.tld
$(local_domains) = $(primary_domain)

log syslog

tls file /etc/certs/$(hostname).crt /etc/certs/$(hostname).key {
    protocols tls1.2 tls1.3
}

auth.pass_table local_authdb {
    table sql_table {
        driver sqlite3
        dsn credentials.db
        table_name passwords
    }
}

storage.imapsql local_mailboxes {
    driver sqlite3
    dsn imapsql.db
}

hostname $(hostname)

msgpipeline local_routing {
    destination_in &local_mailboxes {
        deliver_to &local_mailboxes
    }
    destination postmaster {
        modify {
            replace_rcpt static {
                entry postmaster postmaster@$(primary_domain)
            }
        }
        deliver_to &local_mailboxes
    }
    destination $(local_domains) {
        modify {
            replace_rcpt regexp ".*" "inbox@$(hostname)"
        }
        deliver_to &local_mailboxes
    }

    default_destination {
        reject 550 5.1.1 "User doesn't exist"
    }
}

smtp tcp://0.0.0.0:25 {
    limits {
        all rate 20 1s
        all concurrency 10
    }

    dmarc yes
    check {
        require_mx_record
        dkim
        spf
    }

    source $(local_domains) {
        reject 501 5.1.8 "Use Submission for outgoing SMTP"
    }
    default_source {
        destination postmaster $(local_domains) {
            deliver_to &local_routing
        }
        default_destination {
            reject 550 5.1.1 "User doesn't exist"
        }
    }
}

submission tls://0.0.0.0:465 {
    limits {
        all rate 50 1s
    }

    auth &local_authdb

    source $(local_domains) {
        destination postmaster $(local_domains) {
            deliver_to &local_routing
        }
        default_destination {
            modify {
                dkim $(primary_domain) $(local_domains) default
            }
            deliver_to &remote_queue
        }
    }
    default_source {
        reject 501 5.1.8 "Non-local sender domain"
    }
}

target.remote outbound_delivery {
    limits {
        destination rate 20 1s
        destination concurrency 10
    }
    mx_auth {
        dane
        mtasts {
            cache fs
            fs_dir mtasts_cache/
        }
        local_policy {
            min_tls_level encrypted
            min_mx_level none
        }
    }
}

target.queue remote_queue {
    target &outbound_delivery

    autogenerated_msg_domain $(primary_domain)
    bounce {
        destination postmaster $(local_domains) {
            deliver_to &local_routing
        }
        default_destination {
            reject 550 5.0.0 "Refusing to send DSNs to non-local addresses"
        }
    }
}

imap tls://0.0.0.0:993 {
    auth &local_authdb
    storage &local_mailboxes
}

openmetrics tcp://127.0.0.1:9749 { }
```

EDIT:
I believe it should be fixed in 226ea827d89d0ae419ea348a924570b598d61ac5. Will try to update.
